File: staylio_frontend/common/interfaces/request/SearchRoomRequest.ts
import { RoomStatus } from "@common/enums/RoomStatus";

export interface SearchRoomRequest {
    keyword: string;
    checkInDate: string;
    checkOutDate: string;
    status: RoomStatus;
    adults: number;
    children: number;
    capacity: number;
    page: number;
    size: number;
    minPrice: number;
    maxPrice: number;
    minRating: number;
    sortBy: string;
    direction: string;
}

File: staylio_backend/src/main/java/com/example/staylio_backend/dto/request/SearchRoomRequest.java
package com.example.staylio_backend.dto.request;

import com.example.staylio_backend.model.enums.RoomStatus;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
public class SearchRoomRequest {
    private String keyword;
    private LocalDate checkInDate;
    private LocalDate checkOutDate;
    private Integer adults;
    private Integer children;
    private Integer capacity;
    private RoomStatus status;
    private BigDecimal minPrice;
    private BigDecimal maxPrice;

    private Double minRating;
    private Long provinceId;
    private Long wardId;
    private int page;
    private int size;
    private String sortBy;
    private String direction;
}
